src/ServiceControl/Monitoring/Web/EndpointsMonitoringController.cs

```diff
@@ -29,10 +34,51 @@ internal EndpointsMonitoringController(EndpointInstanceMonitoring monitoring, Ge
         [HttpGet]
         public OkNegotiatedContentResult<EndpointsView[]> Endpoints() => Ok(endpointInstanceMonitoring.GetEndpoints());
 
+        //Added as a way for SP to check if the DELETE operation is supported by the SC API
+        [Route("endpoints")]
+        [HttpOptions]
+        public HttpResponseMessage GetSupportedOperations()
+        {
+            var response = new HttpResponseMessage(HttpStatusCode.NoContent)
+            {
+                Content = new ByteArrayContent(new byte[] { }) //need to force empty content to avoid null reference when adding headers below :(
+            };
+            
+            response.Content.Headers.Allow.Add("GET");
+            response.Content.Headers.Allow.Add("DELETE");
+            response.Content.Headers.Allow.Add("PATCH");
+            response.Content.Headers.Add("Access-Control-Expose-Headers", "Allow");
+            return response;
+        } 
+
+        [Route("endpoints/{endpointId}")]
+        [HttpDelete]
+        public async Task<StatusCodeResult> DeleteEndpoint(Guid endpointId)
+        {
+            var removedFromCache = endpointInstanceMonitoring.RemoveEndpoint(endpointId);
+            if (removedFromCache == null)
+            {
+                return StatusCode(HttpStatusCode.NotFound);
+            }
+
+            await DeletePersistedEndpoint(endpointId).ConfigureAwait(false);
```

Comment:
Should we swap `DeletePersistedEndpoint` and `RemoveEndpoint` from the cache? I think the Raven based is more likely to fail (due to network issues) and that would leave the cache and the storage in an inconsistent state until restart.

Reply:
Sure, why not.